movement/utils/reports.py

```diff
@@ -1,93 +1,67 @@
-
-    This function calculates the number and percentage of NaN points
-    for a given keypoint and individual in the input data. A keypoint
-    is considered NaN if any of its ``space`` coordinates are NaN.
-
-    Parameters
-    ----------
-    data : xarray.DataArray
-        The input data containing ``keypoints`` and ``individuals``
-        dimensions.
-    keypoint : str, optional
-        The name of the keypoint for which to generate the report.
-        If ``None``, it is assumed that the input data contains only
-        one keypoint and this keypoint is used.
-        Default is ``None``.
-    individual : str, optional
-        The name of the individual for which to generate the report.
-        If ``None``, it is assumed that the input data contains only
-        one individual and this individual is used.
-        Default is ``None``.
-
-    Returns
-    -------
-    str
-        A string containing the report.
-
-    """
-    selection_criteria = {}
-    if individual is not None:
-        selection_criteria["individuals"] = individual
-    if keypoint is not None:
-        selection_criteria["keypoints"] = keypoint
-    selected_data = (
-        data.sel(**selection_criteria) if selection_criteria else data
-    )
-    n_nans = selected_data.isnull().any(["space"]).sum(["time"]).item()
-    n_points = selected_data.time.size
-    percent_nans = round((n_nans / n_points) * 100, 1)
-    return f"\n\t\t{keypoint}: {n_nans}/{n_points} ({percent_nans}%)"
+from movement.validators.arrays import validate_dims_coords
 
 
 def report_nan_values(da: xr.DataArray, label: str | None = None) -> str:
-    """Report the number and percentage of keypoints that are NaN.
+    """Report the number and percentage of data points that are NaN.
 
-    Numbers are reported for each individual and keypoint in the data.
+    The number of NaNs are counted for each element along the required
+    ``time`` dimension.
+    If the DataArray has the ``space`` dimension, the ``space`` dimension
+    is reduced by checking if any values in the ``space`` coordinates
+    are NaN, e.g. a 2D point is considered as NaN if any of its x or y
+    coordinates are NaN.
 
     Parameters
     ----------
     da : xarray.DataArray
-        The input data containing ``keypoints`` and ``individuals``
-        dimensions.
+        The input data with ``time`` as a required dimension.
     label : str, optional
         Label to identify the data in the report. If not provided,
-        the name of the DataArray is used as the label.
-        Default is ``None``.
+        the name of the DataArray is used. If the DataArray has no
+        name, "data" is used as the label.
 
     Returns
     -------
     str
         A string containing the report.
 
     """
-    # Compile the report
-    label = label or da.name
+    validate_dims_coords(da, {"time": []})
+    label = label or da.name or "data"
     nan_report = f"\nMissing points (marked as NaN) in {label}"
```

Comment:
To be consistent with the string about "No missing points...", which ends in a period.
```suggestion
    nan_report = f"\nMissing points (marked as NaN) in {label}."
```

Reply:
Could also and in a colon ":" I guess (because the table follows), but it looks weird if it ends with no punctuation mark.

Reply:
Went with ":"